types/validator_set.go

```diff
@@ -471,8 +479,9 @@ func computeNewPriorities(updates []*Validator, vals *ValidatorSet, updatedTotal
 // Expects updates to be a list of updates sorted by address with no duplicates or errors,
 // must have been validated with verifyUpdates() and priorities computed with computeNewPriorities().
 func (vals *ValidatorSet) applyUpdates(updates []*Validator) {
-
 	existing := vals.Validators
+	sort.Sort(ValidatorsByAddress(existing))
```

Comment:
any reason why we sort by address and not power here?

Reply:
the algorithm is faster that way (check out the applyUpdates function body). we sort it by address and go through both arrays. otherwise, we'd have to execute N loops with linear search